ods-devenv/scripts/deploy.sh

```diff
@@ -0,0 +1,1574 @@
+    echo "Create a simple test project to smoke test OpenShift cluster"
+    oc -o yaml new-app php~https://github.com/sandervanvugt/simpleapp --name=simpleapp > s2i.yaml
+    oc create -f s2i.yaml
+    rm s2i.yaml
+}
+
+#######################################
+# install tailor v0.13.1
+# Globals:
+#   n/a
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function download_tailor() {
+    echo "Download tailor"
+    curl -LO "https://github.com/opendevstack/tailor/releases/download/v1.1.0/tailor-linux-amd64"
+    chmod +x tailor-linux-amd64
+    sudo mv tailor-linux-amd64 /usr/bin/tailor
+}
+
+#######################################
+# list versions of software installed in the basic setup
+# Globals:
+#   n/a
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function print_system_setup() {
+    echo "network interfaces: $(ip a)"
+    echo "tailor version: $(tailor version)"
+    echo "oc version: $(oc version)"
+    echo "jq version: $(jq --version)"
+    echo "go version: $(go version)"
+    echo "git version: $(git --version)"
+    echo "docker version: $(docker --version)"
+}
+
+#######################################
+# Retrieve database dump and backup files for Atlassian stack components
+# initialized with timebomb licenses and ODS users from backup in the cloud.
+# If someone would rather start with a clean state and provide licenses and
+# basic configuration manually, this function should be skipped in the setup.
+# This function will try to clean up old states by deleting local folders
+# mapped to docker volumes etc. So, it should not be called in a running
+# system.
+# process.
+# Globals:
+#   n/a
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function prepare_atlassian_stack() {
+    echo "Downloading data dumps for Atlassian stack."
+    pushd "/home/${USER}"
+    curl -LO ${atlassian_mysql_dump_url}
+    curl -LO ${atlassian_jira_backup_url}
+    curl -LO ${atlassian_bitbucket_backup_url}
+    echo "Extracting files"
+    for data_file in bitbucket_data jira_data mysql_data
+    do
+        # cleaning up (stale) files and folders
+        rm -rf "/home/${USER:?}/${data_file:?}"
+        # download and expand archives
+        tar xzf "${data_file}.tar.gz"
+        rm "${data_file}.tar.gz"
+    done
+    popd
+    echo "Finished downloading and extracting Atlassian stack data dumps."
+}
+
+#######################################
+# Start up a containerized mysql instance capable of hosting Atlassian databases
+# for Jira, BitBucket, Confluence.
+# Globals:
+#   atlassian_mysql_container_name
+#   atlassian_mysql_port
+#   atlassian_mysql_version
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function startup_atlassian_mysql() {
+    echo "Setting up a mysql instance for the Atlassian tool suite."
+    docker container run -dp ${atlassian_mysql_port}:3306 \
```

Comment:
Since there are multpitle `docker run` calls, docker compose could be explored/used to manage the docker infrastructure. The code also would become "smaller"

Reply:
@hugowschneider  - we tried, but failed.. we leave it as is :)